aria/orchestrator/execution_plugin/ctx_proxy/server.py

```diff
@@ -150,43 +150,101 @@ def __exit__(self, *args, **kwargs):
         self.close()
 
 
-def _process_ctx_request(ctx, args):
+def _process_ctx_request(ctx, args): # pylint: disable=too-many-branches,too-many-statements
     current = ctx
-    num_args = len(args)
     index = 0
+
+    try:
+        # TODO: should there be a way to escape "=" in case it is needed as real argument?
+        equals_index = args.index('=')
+        if equals_index == 0:
+            raise RuntimeError('The "=" argument cannot be first')
+        if equals_index != len(args) - 2:
+            raise RuntimeError('The "=" argument must be penultimate')
+        modifying = True
+        modifying_key = args[-3]
+        modifying_value = args[-1]
+        args = args[:-3]
+    except ValueError:
+        modifying = False
+        modifying_key = None
+        modifying_value = None
+
+    num_args = len(args)
+
     while index < num_args:
         arg = args[index]
+
+        # Object attribute
         attr = _desugar_attr(current, arg)
-        if attr:
+        if attr is not None:
             current = getattr(current, attr)
-        elif isinstance(current, collections.MutableMapping):
-            key = arg
-            path_dict = _PathDictAccess(current)
-            if index + 1 == num_args:
-                # read dict prop by path
-                value = path_dict.get(key)
-                current = value
-            elif index + 2 == num_args:
-                # set dict prop by path
-                value = args[index + 1]
-                path_dict.set(key, value)
-                current = None
-            else:
-                raise RuntimeError('Illegal argument while accessing dict')
-            break
+
+        # List entry
+        elif isinstance(current, list) and _is_int(arg):
+            current = current[int(arg)]
+
+        # Dict (or dict-like object) value
+        elif hasattr(current, '__getitem__'):
+            if modifying and (not arg in current):
```

Comment:
According to these semantics we lose any knowledge of whether we are accessing a list or a dict